### PR TITLE
[Numpy  fix-doc]modify numpy doc

### PR DESCRIPTION
**Pull request by gyshi**
## Description ##
(Brief description on what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: python/mxnet/_numpy_op_doc.py [hunk 1]
```diff
@@ -32,24 +33,70 @@ def _np_reshape(a, newshape, order='C'):
         an integer, then the result will be a 1-D array of that length.
         One shape dimension can be -1. In this case, the value is
         inferred from the length of the array and remaining dimensions.
-    order : {'C'}, optional
+    order : {'C', 'F', 'A'}, optional
```
Comment (haojin2) [inline python/mxnet/_numpy_op_doc.py]: We do not support 'F' and 'A' orders, why did you add them to the doc?

File: python/mxnet/_numpy_op_doc.py [hunk 2]
```diff
@@ -32,24 +33,70 @@ def _np_reshape(a, newshape, order='C'):
...
+       [4., 5.]])
+
+    You can think of reshaping as first raveling the array (using the given
+    index order), then inserting the elements from the raveled array into the
+    new array using the same kind of index ordering as was used for the
+    raveling.
+
+    >>> np.reshape(a, (2, 3)) # C-like index ordering
+    array([[0., 1., 2.],
+       [3., 4., 5.]])
```
Comment (haojin2) [inline python/mxnet/_numpy_op_doc.py]: Please make sure the alignment looks good on the rendered website.

Comment (gyshi) [reply python/mxnet/_numpy_op_doc.py]: done

File: python/mxnet/_numpy_op_doc.py (hunk 1 above)
Comment (gyshi) [reply python/mxnet/_numpy_op_doc.py]: in the notes ，i have  give a hint

File: python/mxnet/_numpy_op_doc.py [hunk 3]
```diff
@@ -32,24 +33,70 @@ def _np_reshape(a, newshape, order='C'):
...
+     >>> a = np.zeros((10, 2))
+     # A transpose makes the array non-contiguous
+     >>> b = a.T
+     # Taking a view makes it possible to modify the shape without modifying
+     # the initial object.
+
+    >>> a = np.arange(6).reshape((3, 2))
+    >>> a
+    array([[0., 1.],
+       [2., 3.],
```
Comment (reminisce) [inline python/mxnet/_numpy_op_doc.py]: The alignment seems not correct. See the screen shot.
image

File: python/mxnet/_numpy_op_doc.py [hunk 4]
```diff
@@ -32,24 +33,70 @@ def _np_reshape(a, newshape, order='C'):
...
+    new array using the same kind of index ordering as was used for the
+    raveling.
+
+    >>> np.reshape(a, (2, 3)) # C-like index ordering
+    array([[0., 1., 2.],
+       [3., 4., 5.]])
+
+    - order only support C-order
+    - input not support scalar
+    - not support zero-size shape
```
Comment (reminisce) [inline python/mxnet/_numpy_op_doc.py]: This is not true.

File: python/mxnet/_numpy_op_doc.py [hunk 5]
```diff
@@ -32,24 +33,70 @@ def _np_reshape(a, newshape, order='C'):
...
+
+    Examples
     --------
-    ndarray.reshape : Equivalent method.
+    >>> a = np.array([[1,2,3], [4,5,6]])
+    >>> np.reshape(a, 6)
+    array([1., 2., 3., 4., 5., 6.])
+
+    >>> np.reshape(a, (3,-1))       # the unspecified value is inferred to be 2
```
Comment (zoeygxy) [inline python/mxnet/_numpy_op_doc.py]: My output shows float type here.
image

File: python/mxnet/_numpy_op_doc.py [hunk 6]
```diff
@@ -173,3 +220,87 @@ def _np_cumsum(a, axis=None, dtype=None, out=None):
...
+        in the result as dimensions with size one. With this option,
+        the result will broadcast correctly against the input array.
+
+        If the default value is passed, then `keepdims` will not be
+        passed through to the `amax` method of sub-classes of
+        `ndarray`, however any non-default value will be.  If the
+        sub-class' method does not implement `keepdims` any
+        exceptions will be raised.
+
+    initial : scalar, optional
```
Comment (zoeygxy) [inline python/mxnet/_numpy_op_doc.py]: We do not support intial for now. Please specify that users need to pass in "None"

File: python/mxnet/_numpy_op_doc.py (hunk 3 above)
Comment (gyshi) [reply python/mxnet/_numpy_op_doc.py]: done

File: python/mxnet/ndarray/numpy/_op.py [hunk 7]
```diff
@@ -882,3 +941,154 @@ def sqrt(x, out=None, **kwargs):
...
+
+    Equivalent to ``np.sinh(x)/np.cosh(x)``.
+
+    Parameters
+    ----------
+    x : ndarray or scalar.
+        Input array.
+    out : ndarray or None
+        A location into which the result is stored. If provided, it
+        must have a shape that the inputs broadcast to. If not provided
```
Comment (zoeygxy) [inline python/mxnet/ndarray/numpy/_op.py]: ``out`` does not broadcast

File: python/mxnet/numpy/multiarray.py [hunk 8]
```diff
@@ -1318,25 +1318,61 @@ def array(object, dtype=None, ctx=None):
...
-    dtype : str or numpy.dtype, optional
-        An optional value type (default is `numpy.float32`). Note that this
-        behavior is different from NumPy's `ones` function where `float64`
-        is the default value, because `float32` is considered as the default
-        data type in deep learning.
+    shape : int , tuple of ints or list of ints
+        Shape of the new array, e.g., ``(2, 3)`` or ``2``.
+    dtype : data-type, optional
+        The desired data-type for the array, e.g., `numpy.int8`.  Default is
+        `numpy.float64`.
```
Comment (zoeygxy) [inline python/mxnet/numpy/multiarray.py]: Default is float32.

Comment (zoeygxy) [reply python/mxnet/numpy/multiarray.py]:
```
>>> a = np.zeros((2, 3))
>>> a.dtype
<class 'numpy.float32'>
```

File: python/mxnet/symbol/numpy/_symbol.py [hunk 9]
```diff
@@ -1555,4 +1613,156 @@ def sqrt(x, out=None, **kwargs):
...
+    and returns a reference to `out`.  (See Examples)
+
+    - Not support complex computation (like imaginary number)
+
+    >>> np.tanh(np.pi*1j)
+    TypeError: type <type 'complex'> not supported
+
+    Examples
+    --------
+    >>> np.tanh(np.array[0, np.pi]))
```
Comment (zoeygxy) [inline python/mxnet/symbol/numpy/_symbol.py]: Please check parentheses

File: python/mxnet/_numpy_op_doc.py [hunk 10]
```diff
@@ -21,7 +21,8 @@
 
 
 def _np_reshape(a, newshape, order='C'):
```
Comment (zoeygxy) [inline python/mxnet/_numpy_op_doc.py]: Please add parameter ``out`` for ``np.reshape`` . It is the fourth argument.
```
>>> a = np.arange(6)
>>> a
array([0., 1., 2., 3., 4., 5.])
>>> b = np.zeros((3, 2))
>>> np.reshape(a, (3, 2), out=b)
array([[0., 1.],
       [2., 3.],
       [4., 5.]])
>>> b
array([[0., 1.],
       [2., 3.],
       [4., 5.]])
```

File: python/mxnet/ndarray/numpy/_op.py [hunk 11]
```diff
@@ -28,31 +28,70 @@
...
-    dtype : str or numpy.dtype, optional
-        An optional value type. Default is `numpy.float32`. Note that this
-        behavior is different from NumPy's `ones` function where `float64`
-        is the default value, because `float32` is considered as the default
-        data type in deep learning.
+    shape : int , tuple of ints or list of ints
+        Shape of the new array, e.g., ``(2, 3)`` or ``2``.
+    dtype : data-type, optional
+        The desired data-type for the array, e.g., `numpy.int8`.  Default is
+        `numpy.float64`.
```
Comment (zoeygxy) [inline python/mxnet/ndarray/numpy/_op.py]: Default is float32. Please change it in other docs too

File: python/mxnet/_numpy_op_doc.py [hunk 12]
```diff
@@ -173,3 +212,73 @@ def _np_cumsum(a, axis=None, dtype=None, out=None):
...
+
+        If the default value is passed, then `keepdims` will not be
+        passed through to the `amax` method of sub-classes of
+        `ndarray`, however any non-default value will be.  If the
+        sub-class' method does not implement `keepdims` any
+        exceptions will be raised.
+
+    initial :
+        Parameter initial is not supported yet, we will support it in the future.
+        now it must be None.
```
Comment (zoeygxy) [inline python/mxnet/_numpy_op_doc.py]: Now

File: python/mxnet/_numpy_op_doc.py [hunk 13]
```diff
@@ -173,3 +212,73 @@ def _np_cumsum(a, axis=None, dtype=None, out=None):
...
+
+    out : ndarray, optional
```
Comment (zoeygxy) [inline python/mxnet/_numpy_op_doc.py]: Dtype should also be the same: currently we do not support auto type casting like numpy.

File: python/mxnet/_numpy_op_doc.py [hunk 14]
```diff
@@ -173,3 +212,73 @@ def _np_cumsum(a, axis=None, dtype=None, out=None):
...
+        If `axis` is given, the result is an array of dimension
+        ``a.ndim - 1``.
+
+    See Also
+    --------
+    amax     equivalent function
+
+    Notes
+    -----
+    - Not support axis < 0.
```
Comment (zoeygxy) [inline python/mxnet/_numpy_op_doc.py]: “Does not support ....”

File: python/mxnet/_numpy_op_doc.py [hunk 15]
```diff
@@ -173,3 +212,73 @@ def _np_cumsum(a, axis=None, dtype=None, out=None):
...
+    >>> np.max(np.array([[-50], [10]]), axis= -1)
+    i >= 0 && i < ndim(): axis = -1 must be in range [0, 2)
+
+    Examples
+    --------
+    >>> a = np.arange(4).reshape((2,2))
+    >>> a
+    array([[0., 1.],
+          [2., 3.]])
+    >>> np.amax(a)           # Maximum of the flattened array
```
Comment (zoeygxy) [inline python/mxnet/_numpy_op_doc.py]: Shouldn't the examples here be np.max()?

Comment (gyshi) [reply python/mxnet/_numpy_op_doc.py]: they are the same function

Comment (zoeygxy) [reply python/mxnet/_numpy_op_doc.py]: @haojin2

File: python/mxnet/ndarray/numpy/_op.py [hunk 16]
```diff
@@ -864,7 +925,7 @@ def sqrt(x, out=None, **kwargs):
 
     Parameters
     ----------
-    x : ndarray or scalar
+    x : ndarray
```
Comment (hgt312) [inline python/mxnet/ndarray/numpy/_op.py]: scalar is supported

Comment (gyshi) [reply python/mxnet/ndarray/numpy/_op.py]: it's no mine, i rebase from apache/numpy

Comment (zoeygxy) [reply python/mxnet/ndarray/numpy/_op.py]: Be sure to update the numpy branch version before you push next time; the latest version has "scalar" added now.

File: python/mxnet/numpy/multiarray.py [hunk 17]
```diff
@@ -1494,7 +1530,7 @@ def concatenate(seq, axis=0, out=None):
 
     Parameters
     ----------
-    a1, a2, ... : sequence of array_like
+    seq: sequence of ndarray
```
Comment (tingying2020) [inline python/mxnet/numpy/multiarray.py]: There should be one space after seq.

File: python/mxnet/ndarray/numpy/_op.py [hunk 18]
```diff
@@ -28,31 +28,65 @@
...
+    >>> np.zeros((2, 1))
+    array([[0.],
+           [0.]])
+
+    >>> s = (2,2)
+    >>> np.zeros(s)
+    array([[0., 0.],
+           [0., 0.]])
+
+    if you install mxnet gpu version, you can use gpu.
```
Comment (haojin2) [inline python/mxnet/ndarray/numpy/_op.py]: This line is not needed.

Comment (haojin2) [reply python/mxnet/ndarray/numpy/_op.py]: Well, actually it could be useful to have this line, let's still keep it.

File: python/mxnet/ndarray/numpy/_op.py [hunk 19]
```diff
@@ -680,7 +739,7 @@ def linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None, axis
     axis could only be 0 now.
     """
     if isinstance(start, (list, _np.ndarray, NDArray)) or \
-       isinstance(stop, (list, _np.ndarray, NDArray)):
+            isinstance(stop, (list, _np.ndarray, NDArray)):
```
Comment (haojin2) [inline python/mxnet/ndarray/numpy/_op.py]: Why extra indentation here?

File: python/mxnet/ndarray/numpy/_op.py [hunk 20]
```diff
@@ -878,7 +937,160 @@ def sqrt(x, out=None, **kwargs):
...
+    -------
+    y : ndarray or scalar
+       The corresponding hyperbolic tangent values.
+
+    Notes
+    -----
+    If `out` is provided, the function writes the result into it,
+    and returns a reference to `out`.  (See Examples)
+
+    - Not support complex computation (like imaginary number)
```
Comment (haojin2) [inline python/mxnet/ndarray/numpy/_op.py]: "Does not support..."

File: python/mxnet/numpy/multiarray.py [hunk 21]
```diff
@@ -1318,25 +1318,59 @@ def array(object, dtype=None, ctx=None):
...
     Returns
     -------
     out : ndarray
-        Array of zeros with the given shape, dtype, and ctx.
+        Array of zeros with the given shape, dtype, and order.
+
+    Notes
+    -----
+    - Not support ndarray type
+    - Not support custom dtype
```
Comment (haojin2) [inline python/mxnet/numpy/multiarray.py]: What does those two notes mean? Please explain a bit.

Comment (gyshi) [reply python/mxnet/numpy/multiarray.py]: if input is np.array((3,2)) , it will return error!

File: python/mxnet/ndarray/numpy/_op.py (hunk 20 above)
Comment (gyshi) [reply python/mxnet/ndarray/numpy/_op.py]: yes,  not support

Comment (zoeygxy) [reply python/mxnet/ndarray/numpy/_op.py]: What we mean is that grammatically speaking, this sentence should be rephrased as "Does not support complex computation (like imaginary number)". Do not start a sentence with 'Not support...'

Comment (zoeygxy) [reply python/mxnet/ndarray/numpy/_op.py]: Also, you can simply say "Does not support complex number as input". `Complex computation` is an ambiguous expression.

File: python/mxnet/_numpy_op_doc.py (hunk 10 above)
Comment (zoeygxy) [reply python/mxnet/_numpy_op_doc.py]: @gyshi

File: python/mxnet/_numpy_op_doc.py (hunk 5 above)
Comment (zoeygxy) [reply python/mxnet/_numpy_op_doc.py]: @gyshi

File: python/mxnet/_numpy_op_doc.py (hunk 12 above)
Comment (zoeygxy) [reply python/mxnet/_numpy_op_doc.py]: @gyshi  Please captalize the first word in a sentence

File: python/mxnet/_numpy_op_doc.py (hunk 10 above)
Comment (gyshi) [reply python/mxnet/_numpy_op_doc.py]: resolve

File: python/mxnet/_numpy_op_doc.py (hunk 5 above)
Comment (gyshi) [reply python/mxnet/_numpy_op_doc.py]: resolve

File: python/mxnet/_numpy_op_doc.py (hunk 14 above)
Comment (zoeygxy) [reply python/mxnet/_numpy_op_doc.py]: @gyshi

File: python/mxnet/_numpy_op_doc.py (hunk 13 above)
Comment (gyshi) [reply python/mxnet/_numpy_op_doc.py]: resolve,

File: python/mxnet/ndarray/numpy/_op.py (hunk 18 above)
Comment (gyshi) [reply python/mxnet/ndarray/numpy/_op.py]: resolve

File: python/mxnet/symbol/numpy/_symbol.py (hunk 9 above)
Comment (zoeygxy) [reply python/mxnet/symbol/numpy/_symbol.py]: @gyshi You have omitted one left parenthesis here

Comment (gyshi) [reply python/mxnet/symbol/numpy/_symbol.py]: resolve

File: python/mxnet/ndarray/numpy/_op.py (hunk 16 above)
Comment (gyshi) [reply python/mxnet/ndarray/numpy/_op.py]: resolve

File: python/mxnet/ndarray/numpy/_op.py (hunk 19 above)
Comment (gyshi) [reply python/mxnet/ndarray/numpy/_op.py]: resolve

File: python/mxnet/numpy/multiarray.py (hunk 21 above)
Comment (gyshi) [reply python/mxnet/numpy/multiarray.py]: resolve

File: python/mxnet/symbol/numpy/_symbol.py [hunk 22]
```diff
@@ -1549,10 +1613,161 @@ def sqrt(x, out=None, **kwargs):
...
+
+    The ceil of the ndarray `x` is the smallest integer `i`, such that
+    `i >= x`.  It is often denoted as :math:`\lceil x \rceil`.
+
+    Parameters
+    ----------
+    x : _Symbol or scalar
+        Input array.
+    out : _Symbol or None
+        A location into which the result is stored. If provided, it
```
Comment (ckt624) [inline python/mxnet/symbol/numpy/_symbol.py]: Based on templates:
out : _Symbol or None
        Dummy parameter to keep the consistency with the ndarray counterpart.

File: python/mxnet/symbol/numpy/_symbol.py [hunk 23]
```diff
@@ -1549,10 +1613,161 @@ def sqrt(x, out=None, **kwargs):
...
+
+    Calculates ``log(1 + x)``.
+
+    Parameters
+    ----------
+    x : _Symbol or scalar
+        Input array.
+    out : _Symbol or None
+        A location into which the result is stored. If provided, it
+        must have a shape that the inputs fill into. If not provided
```
Comment (ckt624) [inline python/mxnet/symbol/numpy/_symbol.py]: Based on templates:
out : _Symbol or None
        Dummy parameter to keep the consistency with the ndarray counterpart.

File: python/mxnet/symbol/numpy/_symbol.py [hunk 24]
```diff
@@ -1549,10 +1613,161 @@ def sqrt(x, out=None, **kwargs):
...
+
+    Parameters
+    ----------
+    x : _Symbol
+        Input array.
+    out : _Symbol or None
+        A location into which the result is stored. If provided, it
+        must have a shape that the inputs fill into. If not provided
+        or None, a freshly-allocated array is returned. The dtype of the
+        output and input must be the same.
```
Comment (ckt624) [inline python/mxnet/symbol/numpy/_symbol.py]: Based on templates:
out : _Symbol or None
        Dummy parameter to keep the consistency with the ndarray counterpart.

Comment (gyshi) [reply python/mxnet/symbol/numpy/_symbol.py]: resolve

File: python/mxnet/symbol/numpy/_symbol.py (hunk 23 above)
Comment (gyshi) [reply python/mxnet/symbol/numpy/_symbol.py]: resolve

File: python/mxnet/_numpy_op_doc.py (hunk 5 above)
Comment (zoeygxy) [reply python/mxnet/_numpy_op_doc.py]: Have you tested this?

File: python/mxnet/_numpy_op_doc.py (hunk 13 above)
Comment (zoeygxy) [reply python/mxnet/_numpy_op_doc.py]: @gyshi Please add this in the `note` area. This is different from native numpy. I would also suggest giving a note here in `out` parameter description.

File: python/mxnet/symbol/numpy/_symbol.py (hunk 9 above)
Comment (zoeygxy) [reply python/mxnet/symbol/numpy/_symbol.py]: @gyshi

**Comment by gyshi** [comment]
@reminisce  my pr have some error, i think it is not my error, for example:  FullyQualifiedErrorId : WebException, can you see my error messages.

File: python/mxnet/_numpy_op_doc.py [hunk 25]
```diff
@@ -22,37 +22,73 @@
...
         inferred from the length of the array and remaining dimensions.
-    order : {'C'}, optional
+    order : 'C', optional
         Read the elements of `a` using this index order, and place the
         elements into the reshaped array using this index order.  'C'
         means to read / write the elements using C-like index order,
         with the last axis index changing fastest, back to the first
-        axis index changing slowest. Other order types such as 'F'/'A'
-        may be added in the future.
+        axis index changing slowest. We will add 'F'/'A' order types
```
Comment (aaronmarkham) [inline python/mxnet/_numpy_op_doc.py]: Would reword to not make promises. Link to the issue/proposal doc.

File: python/mxnet/_numpy_op_doc.py [hunk 26]
```diff
@@ -22,37 +22,73 @@
...
+    Notes
+    -----
+    It is not always possible to change the shape of an array without
+    copying the data. If you want an error to be raised when the data is copied,
+    you should assign the new shape to the shape attribute of the array::
+
+     >>> a = np.zeros((10, 2))
+     # A transpose makes the array non-contiguous
+     >>> b = a.T
+     # Taking a view makes it possible to modify the shape without modifying
```
Comment (aaronmarkham) [inline python/mxnet/_numpy_op_doc.py]: Seems like you're going to show an example of "taking a view"... but then it isn't clear to me in the example what a view is. Is this comment associated with line 60 or 64?

File: python/mxnet/_numpy_op_doc.py [hunk 27]
```diff
@@ -22,37 +22,73 @@
...
+     # the initial object.
+
+    >>> a = np.arange(6).reshape((3, 2))
+    >>> a
+    array([[0., 1.],
+           [2., 3.],
+           [4., 5.]])
+
+    You can think of reshaping as first raveling the array (using the given
```
Comment (aaronmarkham) [inline python/mxnet/_numpy_op_doc.py]: I'm not sure about the use of "raveling" here. Its meanings are sort of ambiguous and rely on context.

I realize you're copying from Numpy docs. That's doesn't mean that it is perfectly clear. 

Are you adding `np.ravel` too?

File: python/mxnet/_numpy_op_doc.py (hunk 25 above)
Comment (gyshi) [reply python/mxnet/_numpy_op_doc.py]: done
